### PR TITLE
fix(rnd): Prevent boolean with no default value on AGPT-builder

### PR DESCRIPTION
**Pull request by majdyz**
### Background

Boolean without default value is a UX problem. It's currently displayed as a toggle and it has no way to describe the `null` value.
So we need to prevent blocks from introducing a nullable boolean.

### Changes 🏗️

Add explicit check to prevent nullable boolean. Fix existing block field that has nullable boolean.

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/search.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ class GetOpenWeatherMapBlock(Block, GetRequest):
     class Input(BlockSchema):
         location: str
         api_key: BlockSecret = SecretField(key="openweathermap_api_key")
-        use_celsius: bool
+        use_celsius: bool = True
```
Comment (majdyz) [inline rnd/autogpt_server/autogpt_server/blocks/search.py:138]: I'm sorry for being anti-US here 🦅 but we had to choose a value.

File: rnd/autogpt_builder/src/components/edit/control/BlocksControl.tsx [hunk 2]
```diff
@@ -36,8 +36,12 @@ export const BlocksControl: React.FC<BlocksControlProps> = ({
...
   const [searchQuery, setSearchQuery] = useState("");
 
-  const filteredBlocks = blocks.filter((block: Block) =>
-    block.name.toLowerCase().includes(searchQuery.toLowerCase()),
+  const filteredBlocks = blocks.filter(
+    (block: Block) =>
+      block.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
+      beautifyString(block.name)
+        .toLowerCase()
+        .includes(searchQuery.toLowerCase()),
```
Comment (majdyz) [inline rnd/autogpt_builder/src/components/edit/control/BlocksControl.tsx:44]: Indirectly related: block is not searchable using its beatified name. e.g: I can't find the GetOpenWeatherMapBlock using `weather map` query.

File: rnd/autogpt_server/autogpt_server/blocks/search.py (hunk 1 above)
Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/blocks/search.py:138]: Screenshot 2024-08-23 at 4 33 00 PM
i think this was the right choice 😅

**Review by aarushik93 — APPROVED** [review]

File: rnd/autogpt_server/autogpt_server/blocks/search.py (hunk 1 above)
Comment (ntindle) [reply rnd/autogpt_server/autogpt_server/blocks/search.py:138]: Heartbreaking

**Review by ntindle — APPROVED** [review]

**Pull request merged**